sound/soc/sof/ipc4-pcm.c

```diff
@@ -106,6 +276,14 @@ static int sof_ipc4_trigger_pipelines(struct snd_soc_component *component,
 		pipeline->state = state;
 	}
 
+	if (state == SOF_IPC4_PIPE_RUNNING) {
+		spcm->stream[substream->stream].stream_start_wclk =
+			SOF_IPC4_INVALID_STREAM_POSITION;
+		/* wait for all components started */
+		schedule_delayed_work(&spcm->stream[substream->stream].wclk_calibration,
+				      msecs_to_jiffies(SOF_IPC4_WCLK_CALIBRATION_DELAY));
```

Comment:
the comment doesn't help.

What do you mean by 'all components'?

Also the delayed work is just a delayed work, there's nothing that waits for resources to be available.

What is this trying to accomplish? Additional explanations are needed here.

Reply:
sure, I will add

Reply:
take SDW as a example, all components means : sdw controller, codec and audio dsp. If sdw controller and codec are not actived, dma engine can't copy data to controller so we can't get correct dma counter